### PR TITLE
Fix Pipeline & upgraded Node to 12.14.1 

### PR DESCRIPTION
**Pull request by johnoneill46**
- Upgraded Node to "12.14.1" from "10.15.2""
- Upgraded relevant versions to ensure compatibility with Node 12
- Increased test timeouts to ensure a more stable build
- Removed unused file: "app/dataStructure.json"

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,9 +25,9 @@
     "test:smoke": "./bin/run-smoke-tests.sh",
     "test:functional": "./bin/run-functional-tests.sh",
     "test:crossbrowser": "./bin/run-crossbrowser-tests.sh",
-    "test-unit": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. mocha './app/**/*.test.js' --reporter spec --recursive -t 10000",
-    "test-unit-ide": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. mocha $NODE_DEBUG_OPTION './app/**/*.test.js' --reporter spec --recursive",
-    "test-validation": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. mocha test/validation/ --reporter spec --recursive --timeout 15000",
+    "test-unit": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. mocha './app/**/*.test.js' --reporter spec --recursive -t 15000 --exit",
+    "test-unit-ide": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. mocha $NODE_DEBUG_OPTION './app/**/*.test.js' --reporter spec --recursive --exit",
+    "test-validation": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. mocha test/validation/ --reporter spec --recursive --timeout 15000 --exit",
```
Comment (johnoneill46) [inline package.json]: the "--exit" is required as we updated from Mocha 3 to Mocha 5 which now no longer automatically kill the tests

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "start-prod": "NODE_PATH=. grunt start-prod",
     "setup-assets": "NODE_PATH=. grunt setup-assets",
     "setup": "NODE_PATH=. grunt setup-prod",
-    "test": "NODE_PATH=. LOG_LEVEL=OFF NODE_ENV=testing nyc mocha 'app/**/*.test.js' -t 10000",
+    "test": "FEATURE_IDAM=false NODE_PATH=. LOG_LEVEL=OFF NODE_ENV=testing nyc mocha 'app/**/*.test.js' -t 15000 --exit",
```
Comment (johnoneill46) [inline package.json]: Temporary fix to fix tests failing. Turns out our migration to AKS has broken how feature toggles are set on the environments. This also means our unit tests rely on actual env variables which isn't good so will fix those after this

File: README.md [hunk 3]
```diff
@@ -12,6 +12,23 @@ brew install yarn
 brew install redis
 ```
 
+If it's your first time running the app then run this command to setup the Grunt config files:
+
+```
+yarn setup
+```
+
+If it's your first time running the app then run this command to setup the Grunt config files:
```
Comment (jvictorcf) [inline README.md]: duplicate section

File: package.json (hunk 2 above)
Comment (jvictorcf) [reply package.json]: Are you planning to merge this?

File: infrastructure/variables.tf [hunk 4]
```diff
@@ -289,7 +289,7 @@ variable "webchat_button_service_closed" {
 }
 
 variable "node_version" {
-  default = "10.15.2"
+  default = "12.14.1"
```
Comment (jvictorcf) [inline infrastructure/variables.tf]: Isn't there an equivalent field in the charts folder?

File: package.json (hunk 2 above)
Comment (johnoneill46) [reply package.json]: The fix has already been merged in a separate PR as it was the only way to get the builds green (According to Tony anyway) so my initial comment is now redundant

**Pull request merged**